docs/development/journalist_api.rst

```diff
@@ -0,0 +1,414 @@
+Journalist Interface API
+========================
+
+This document describes the endpoints for SecureDrop's Journalist Interface
+API.
+
+Versioning
+~~~~~~~~~~
+
+The API is versioned and we are currently using version 1. This is set via the
+base URL, which is:
+
+.. code:: sh
+
+  /api/v1/
+
+Content Type
+~~~~~~~~~~~~
+
+Clients should send the following headers:
+
+.. code:: sh
+
+  'Accept': 'application/json',
+  'Content-Type': 'application/json'
+
+Authentication
+~~~~~~~~~~~~~~
+
+``POST /api/v1/token`` to get a token with the username, password, and 2FA
```

Comment:
That is a good idea! The thing is: we're not actually tracking these tokens server-side, so this wouldn't actually do anything i.e. the token is not really invalidated, only the expiry will invalidate the token.

Aside: we _could_ integrate e.g. [Flask-Session](https://pythonhosted.org/Flask-Session/) in the future so we _can_ do this (i.e. we maintain a blacklist of revoked tokens)

In the meantime, I think reasonable behavior is for clients to logout by simply deleting the token.

Reply:
I might be wrong here, but isn't `Flask-Session` meant just for session *cookies*? Like all we need is a DB table of `(journalist_id, token, expiration)`, no? I know this goes into rolling code we need to maintain, but after a few minutes of browsing, I didn't see any auth token libs out there that were't JWT or cookies. If we end up maintaining a blacklist of cookies, why not just go the otherway and maintain a whitelist only to make things easier?

Reply:
Yeah we just need a server-side data store: the question is if that's something we want to implement in this PR (I vote no if we can avoid it)

Reply:
Ok that's fair. Maybe we can just trade degraded UX to get this in since no one is using it yet, and then we can work out how to renew/revoke cookies.